File: python/src/main/python/pygw/index/spatial_index_builder.py
#
# Copyright (c) 2013-2020 Contributors to the Eclipse Foundation

#
# See the NOTICE file distributed with this work for additional information regarding copyright
# ownership. All rights reserved. This program and the accompanying materials are made available
# under the terms of the Apache License, Version 2.0 which accompanies this distribution and is
# available at http://www.apache.org/licenses/LICENSE-2.0.txt
#===============================================================================================

from pygw.config import geowave_pkg

from .index_builder import IndexBuilder
from .index import Index

class SpatialIndexBuilder(IndexBuilder):
    """
    Index builder for a spatial index.
    """

    def __init__(self):
        super().__init__(geowave_pkg.core.geotime.index.api.SpatialIndexBuilder())

    def set_include_time_in_common_index_model(self, include):
        """
        Sets whether or not to include time in the common index model.  This can be
        used to speed up queries that may involve temporal constraints.

        Args:
            include (bool): Whether or not to include time in the common index model.
        Returns:
            This index builder.
        """
        self._java_ref.setIncludeTimeInCommonIndexModel(include)
        return self

File: core/geotime/src/main/java/org/locationtech/geowave/core/geotime/index/api/SpatialIndexBuilder.java
/**
 * Copyright (c) 2013-2022 Contributors to the Eclipse Foundation
 *
 * <p> See the NOTICE file distributed with this work for additional information regarding copyright
 * ownership. All rights reserved. This program and the accompanying materials are made available
 * under the terms of the Apache License, Version 2.0 which accompanies this distribution and is
 * available at http://www.apache.org/licenses/LICENSE-2.0.txt
 */
package org.locationtech.geowave.core.geotime.index.api;

import javax.annotation.Nullable;
import org.locationtech.geowave.core.geotime.index.SpatialDimensionalityTypeProvider;
import org.locationtech.geowave.core.geotime.index.SpatialOptions;
import org.locationtech.geowave.core.store.api.Index;
import org.locationtech.geowave.core.store.index.BaseIndexBuilder;

public class SpatialIndexBuilder extends BaseIndexBuilder<SpatialIndexBuilder> {
  private final SpatialOptions options;

  public SpatialIndexBuilder() {
    super();
    options = new SpatialOptions();
  }

  public SpatialIndexBuilder setIncludeTimeInCommonIndexModel(final boolean storeTime) {
    options.storeTime(storeTime);
    return this;
  }

  public SpatialIndexBuilder setGeometryPrecision(@Nullable final Integer precision) {
    options.setGeometryPrecision(precision);
    return this;
  }

  public SpatialIndexBuilder setCrs(final String crs) {
    options.setCrs(crs);
    return this;
  }

  @Override
  public Index createIndex() {
    return createIndex(SpatialDimensionalityTypeProvider.createIndexFromOptions(options));
  }
}
